sdk/identity/azure-identity/azure/identity/_internal/interactive.py

```diff
@@ -79,7 +84,7 @@ def _build_auth_record(response):
         six.raise_from(auth_error, ex)
 
 
-class InteractiveCredential(MsalCredential):
+class InteractiveCredential(MsalCredential, ABC):
```

Comment:
What's the reasoning for making InteractiveCredential an ABC instead of MsalCredential? I'm not too familiar with the purpose of inheriting from ABC

Reply:
A class inheriting ABC can only be instantiated if it implements every method the ABC decorates with `abstractmethod`. That's useful here because it enables InteractiveCredential.get_token to consolidate common code and do most of the work for subclasses while ensuring you get a runtime error when a subclass doesn't implement the one method that differs between authentication flows. With this change MsalCredential loses its only abstract method and thus its reason for inheriting ABC. InteractiveCredential still has an abstract method but was inheriting ABC through MsalCredential. So, now it must inherit ABC directly.